docs/data-import-and-processing.adoc

```diff
@@ -62,31 +62,30 @@ This means that you can have `update_other` turned on and `update_parallels` dis
 
 == Synchronization
 
-The external data can be changed or deleted at any time, so the synchronization process has to account for that.
-KOS(api) does not provide any diffs or changesets, so full synchronization has to be performed every time an update is required.
-Synchronization is usually scheduled multiple times a day, so that the most recent data are provided.
+The external data can be changed or deleted any time, so the synchronization process has to account for that.
+KOS(api) does not provide any diffs or changesets, so we need to performa a full synchronization every time an update is required.
+Synchronization happens multiple times a day to provide up-to-date data.
 
-When synchronizing with existing database data, events are generated as usual and when saving them, an attempt is made to match them with the existing events in the database.
+We synchronize existing database data as follows: events are generated as usual. When the generated events are saved to dabase, an attempt is made to match them with the existing events in the database.
 If a match is found and the data is the same, there is no need to update anything.
-In a case there is a match and the event data differs, existing event data in the database had to be updated.
-And in a case there is no match at all, a new event with generated data is created.
+If there is a match and the event data differs, existing event in the database is updated.
+In case there is no match at all, a new event is created.
 
-Event matching is designed so that it provides the most stable matching possible (meaning it does not generate new events when not necessary).
-For data sources which have one-to-one mapping with resulting events (i.e. exams, course events) this is pretty easy.
-The source entity ID (for example an exam ID) is saved in the generated event and in the consecutive runs a match is performed against it.
+Event matching strives to be stable, i.e. it aims to update existing events instead of erasing and generating new events. 
```

Comment:
Maybe replace `erasing and generating` with `erasing and creating` here. Events are generated always during the import and then they are either saved as new records or update existing ones.